tests/io/v3/vm/agent_groups/test_agent_groups_api.py

```diff
@@ -0,0 +1,242 @@
+    '''
+    Test case for create method
+    '''
+    name: str = 'test'
+    test_response: dict = {
+        "agent_groups": [
+            {
+                "id": "ef62870e-fe2f-4ba9-98b7-43d3a53ffe85",
+                "name": name,
+                "creation_date": 1635756224,
+                "last_modification_date": 1635756224,
+                "timestamp": 1635756224,
+                "shared": 1,
+                "owner": {
+                    "name": "system",
+                    "id": "3bfcfb11-6c12-405b-b7ba-bbc705cd2a6e"
+                },
+                "user_permissions": 128,
+                "agents_count": 0
+            }
+        ]
+    }
+    responses.add(
+        responses.POST,
+        re.compile(f'{BASE_URL}'),
+        match=[responses.matchers.json_params_matcher({'name': name})],
+        json=test_response
+    )
+    res = api.v3.vm.agent_groups.create(name)
+    assert isinstance(res, dict)
+    assert res['agent_groups'][0]['name'] == name
+
+
+@responses.activate
+def test_delete(api):
+    '''
+    Test case for delete method
+    '''
+    group_id: UUID = 'e069b272-ed76-487a-8cf9-1c32836698b7'
+    responses.add(
+        responses.DELETE,
+        re.compile(f'{BASE_URL}/{group_id}'),
+        status=200
+    )
+    res = api.v3.vm.agent_groups.delete(group_id)
+    assert res is None
+
+
+@responses.activate
+def test_delete_agent_with_single_agent_id(api):
+    '''
+    Test case for delete_agent method
+    '''
+    group_id: UUID = '2b2db604-5d92-11ec-bf63-0242ac130002'
+    agent_id: UUID = '3f8eed68-5d95-11ec-bf63-0242ac130002'
+    responses.add(
+        responses.DELETE,
+        re.compile(f'{BASE_URL}/{group_id}/agents/{agent_id}')
+    )
+    res = api.v3.vm.agent_groups.delete_agent(group_id, agent_id)
+    assert res is None
+
+
+@responses.activate
+def test_delete_agent_with_multiple_agent_id(api):
+    '''
+    Test case for delete_agent method
+    '''
+    group_id: UUID = 'fs252fdg-4b7c-4d2b-99a1-dvsdsv4242vf'
+    payload = {
+        'items': [
+            i for i in (
+                '57b74c0a-5d95-11ec-bf63-0242ac130002',
+                '57b74e58-5d95-11ec-bf63-0242ac130002',
+                '57b74f66-5d95-11ec-bf63-0242ac130002'
+            )
+        ]
+    }
+    test_repsonse = {
+        'task_id': 'c26d637e-8533-411b-920c-5f49faeb270d',
+        'container_id': 'cfdabb09-6aef-481d-b28f-aecb1c38f297',
+        'status': 'NEW',
+        'message': 'Beginning bulk removeFromGroup operation'
+    }
+    responses.add(
+        responses.POST,
+        re.compile(f'{BASE_URL}/{group_id}/agents/_bulk/remove'),
+        match=[responses.matchers.json_params_matcher(payload)],
+        json=test_repsonse
+    )
+    res = api.v3.vm.agent_groups.delete_agent(
+        group_id,
+        '57b74c0a-5d95-11ec-bf63-0242ac130002',
+        '57b74e58-5d95-11ec-bf63-0242ac130002',
+        '57b74f66-5d95-11ec-bf63-0242ac130002'
+    )
+    assert isinstance(res, dict)
```

Comment:
We should try to compare a couple of keys as well.

Reply:
Added.

Reply:
@milanaghera-tenable it should be 
assert test_response['container_id'] == res['container_id']